package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -92,39 +89,105 @@ define([
     },
 
     /**
-     * Orchestrates the rendering of the visualization.
+     * Orchestrates the rendering of the visualization and is meant to be invoked by the container.
      *
-     * This method executes [_render]{@link pentaho.visual.base.View#_render}
-     * asynchronously and is meant to be invoked by the container.
+     * Executes [_update]{@link pentaho.visual.base.View#_update} asynchronously in
+     * the will/did/rejected event loop associated with an update of a visualization,
+     * and also creates the visualization DOM element the first time it successfully updates.
+     *
+     * In order to get the visualization DOM element,
+     * listen to the {@link pentaho.visual.base.events.DidCreate|"did:create"} event.
      *
      * @return {Promise} A promise that is fulfilled when the visualization
      * is completely rendered. If the visualization is in an invalid state, the promise
      * is immediately rejected.
+     *
+     * @fires "will:update"
+     * @fires "did:update"
+     * @fires "rejected:update"
+     * @fires "did:create"
      */
-    render: function() {
-      var me = this;
+    update: function() {
```

Comment:
Note that the updating phase finishes strictly before the DidUpdate/RejectedUpdate events, so that these can synchronously emit new updates in response, if desired (but after DidCreate).